OloEngine/src/OloEngine/Core/TaskTag.cpp

```
@@ -0,0 +1,244 @@
+	bool GIsGameThreadIdInitialized = false;
+	bool GIsRunningRHIInSeparateThread = false;
+	bool GIsRunningRHIInDedicatedThread = false;
+
+	//////////////////////////////////////////////////////////////////////////
+	// FTaskTagScope implementation
+
+	thread_local ETaskTag FTaskTagScope::s_ActiveTaskTag = ETaskTag::EStaticInit;
+	std::atomic<i32> FTaskTagScope::s_ActiveNamedThreads{0};
+
+	// Store the thread ID of the thread that ran static initialization
+	static u32 s_StaticInitThreadId = FPlatformTLS::GetCurrentThreadId();
```

Clang (14.0.6)

[warning] variable 's_StaticInitThreadId' is non-const and globally accessible, consider making it const

(cppcoreguidelines-avoid-non-const-global-variables)